core/client/tests/unit/components/gh-profile-image-test.js

```diff
@@ -28,7 +28,10 @@ describeComponent(
                 testEmail = 'test@example.com',
                 style, size;
 
-            component.set('email', testEmail);
+            Ember.run(function () {
+                component.set('email', testEmail);
+            });
+
```

Comment:
If something is hard to test it usually highlights that improvements can be made to the design :wink: 

First, I'd drop the unit test and switch to an integration test as it allows you to test the component the way it's used rather than digging into it's implementation, check the `core/client/tests/integration/components` directory for some examples.

I think there are two key changes that could be made to the component that will ease testing:

1. Have the component communicate to the outside world. It should be possible to have the component call `sendAction('gravatarChanged', url)` (`sendAction` is a no-op inside a component if an action wasn't passed in)
2. Make the timeout configurable so it's possible to do `{{gh-profile-image email=email debounce=50}}`. This has the added benefit that we aren't slowing our test suite down unnecessarily :smile: 

With the above changes, in an integration test you could do something like the following (not tested, just an idea, especially chai's `closeTo` method):

```
it('eventually sets the gravatar image with a valid email', function (done) {
    let email = 'test@example.com',
        emailmd5 = md5(email),
        starttime;

    // stub the gravatarChanged action
    this.on('gravatarChanged', (gravatarURL) => {
        let expectedURL = `http://gravatar.com/avatar/${md5email}?size=250&d=blank`,
            endtime = new Date(),
            timediff = endtime - starttime;

        // check the debounce is in the correct range
        expect(timediff, 'time until action fired')
            .to.be.closeTo(100, 20);

        // it passes gravatar URL to action
        expect(gravatarURL, 'url passed to gravatarChanged')
            .to.equal(expectedURL)

        // it sets the background style
        expect(this.$('.gravatar-img').attr('style'), '.gravatar-img style attr')
            .to.equal(`background-image: ${expectedURL}`);

        // finish the async test
        done();
    });

    // start with a non-email
    this.set('email', 'unknown');

    // render the component
    this.render(hbs`
        {{#gh-profile-image email=email debounce=100 gravatarChanged="gravatarChanged"}}
    `);

    // simulate a new value passed in
    run(() => {
        starttime = new Date();
        this.set('email', email);
    });
});
```